spec/rspec/rails/matchers/send_email_spec.rb

```diff
@@ -0,0 +1,168 @@
+    }.to raise_error(RSpec::Expectations::ExpectationNotMetError, <<~MSG.strip)
+      No matching emails were sent.
+
+      The following emails were sent:
+      - subject: Test email, from: ["from@example.com"], to: ["to@example.com"], cc: ["cc@example.com"], bcc: ["bcc@example.com"]
+    MSG
+  end
+
+  it "fails with a clear message when no emails were sent" do
+    expect {
+      expect { }.to send_email
+    }.to raise_error(RSpec::Expectations::ExpectationNotMetError, <<~MSG.strip)
+      No matching emails were sent.
+
+      There were no any emails sent inside the expectation block.
+    MSG
+  end
+
+  it "fails with a clear message for negated version" do
+    expect {
+      expect { mailer.test_email.deliver_now }.to_not send_email(from: "from@example.com")
+    }.to raise_error(RSpec::Expectations::ExpectationNotMetError, "Expected not to send an email but it was sent.")
+  end
+
+  it "fails for multiple matches" do
+    expect {
+      expect { 2.times { mailer.test_email.deliver_now } }.to send_email(from: "from@example.com")
+    }.to raise_error(RSpec::Expectations::ExpectationNotMetError, <<~MSG.strip)
+      More than 1 matching emails were sent.
+
+      The following emails were sent:
+      - subject: Test email, from: ["from@example.com"], to: ["to@example.com"], cc: ["cc@example.com"], bcc: ["bcc@example.com"]
+      - subject: Test email, from: ["from@example.com"], to: ["to@example.com"], cc: ["cc@example.com"], bcc: ["bcc@example.com"]
+    MSG
+  end
+
+  context "with compound matching" do
+    it "works when both matchings pass" do
+      expect {
+        expect {
+          mailer.test_email.deliver_now
+        }.to send_email(to: "to@example.com").and send_email(from: "from@example.com")
+      }.to_not raise_error
+    end
+
+    it "works when first matching fails" do
+      expect {
+        expect {
+          mailer.test_email.deliver_now
+        }.to send_email(to: "no@example.com").and send_email(to: "to@example.com")
+      }.to raise_error(RSpec::Expectations::ExpectationNotMetError, <<~MSG.strip)
+        No matching emails were sent.
+
+        The following emails were sent:
+        - subject: Test email, from: ["from@example.com"], to: ["to@example.com"], cc: ["cc@example.com"], bcc: ["bcc@example.com"]
+      MSG
+    end
+
+    it "works when second matching fails" do
+      expect {
+        expect {
+          mailer.test_email.deliver_now
+        }.to send_email(to: "to@example.com").and send_email(to: "no@example.com")
+      }.to raise_error(RSpec::Expectations::ExpectationNotMetError, <<~MSG.strip)
+        No matching emails were sent.
+
+        The following emails were sent:
+        - subject: Test email, from: ["from@example.com"], to: ["to@example.com"], cc: ["cc@example.com"], bcc: ["bcc@example.com"]
+      MSG
+    end
+  end
+
+  context "with a custom negated version defined" do
+    before { RSpec::Matchers.define_negated_matcher :not_send_email, :send_email }
```

Comment:
```suggestion
    RSpec::Matchers.define_negated_matcher :not_send_email, :send_email
```

Reply:
It also pollutes the global scope. Changing it to use `around`